src/task.c

```diff
@@ -395,6 +325,154 @@ task_thread_create(const char            *name,
    return STATUS_OK;
 }
 
+/******************************************************************************
+ * Registering/deregistering threads that were not started by the task
+ * system.
+ ******************************************************************************/
+
+/*
+ * task_register_thread(): Register this new thread with the task system.
+ *
+ * This is for use by threads created outside of the splinter system.
+ *
+ * Registration implies:
+ *  - Acquire a new thread ID (index) for this to-be-active thread
+ *  - Allocate scratch space for this thread, and track this space..
+ */
+platform_status
+task_register_thread(task_system *ts,
+                     uint64       scratch_size,
+                     const char  *file,
+                     const int    lineno,
+                     const char  *func)
+{
+   threadid thread_tid;
+
+   debug_code(thread_tid = platform_get_tid());
+   debug_assert(
+      thread_tid == INVALID_TID,
+      "[%s:%d::%s()] Attempt to register already registered thread %lu\n",
+      file,
+      lineno,
+      func,
+      thread_tid);
+
+   thread_tid = allocate_threadid(ts);
+   if (thread_tid == INVALID_TID) {
+      return STATUS_NO_SPACE;
+   }
+   platform_set_tid(thread_tid);
```

Comment:
Suggestion for small cleanup: If you could move this line to just before the call to `task_run_thread_hooks()` on L379, then you could remove the call to `platform_set_tid(INVALID_TID);` on L372 below.

All this business of `set_tid()` to some valid value and then to back it out in case of an error could be simplified.